### PR TITLE
fixes #7067: references to the product name are now populated automatically, based on the content of the 'NAME' file.

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by dmitri-d** [comment]
rebased; updated to make smart proxy product name configurable.

**Comment by domcleal** [comment]
I really have no interest in maintaining support for branding forks of Foreman, big :-1:

**Comment by lzap** [comment]
What do you mean by that? People can fork anytime if they like. This can make it easier but I doubt it initiates a single one.

File: app/views/about/index.html.erb [hunk 1]
```diff
@@ -110,7 +110,7 @@
         <li><%= link_to _("First use instructions"), :controller => '/dashboard', :action => 'welcome' %></li>
       </ul>
       <h6><%= _("IRC") %></h6>
-      <p><%= (_("You can find The Foreman on the %{freenode} (irc.freenode.net) network.  For general support, please visit #theforeman and for development specific related chat, please visit #theforeman-dev.") % {:freenode => link_to("freenode", "http://www.freenode.net", :rel => "external")}).html_safe  %></p>
+      <p><%= (_("You can find The %{name} on the %{freenode} (irc.freenode.net) network.  For general support, please visit #theforeman and for development specific related chat, please visit #theforeman-dev.") % {:name => SETTINGS[:name], :freenode => link_to("freenode", "http://www.freenode.net", :rel => "external")}).html_safe  %></p>
```
Comment (cliffy94) [inline app/views/about/index.html.erb:113]: Bad change here. Suggest to leave this bit alone, OR needs work to make more configurable the strings, or don't touch.

File: app/views/common/500.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 <%= alert :class => 'alert-block alert-danger base in fade', :text => h(exception.message) %>
 
 <p>
-  <%= _("If you feel this is an error with Foreman itself, please open a new issue with") %> <%= link_to _("Foreman ticketing system"), "http://theforeman.org/projects/foreman/issues", :rel => "external" %>,
+  <%= _("If you feel this is an error with %s itself, please open a new issue with") % SETTINGS[:name] %> <%= link_to _("Foreman ticketing system"), "http://theforeman.org/projects/foreman/issues", :rel => "external" %>,
```
Comment (cliffy94) [inline app/views/common/500.html.erb:7]: I don't agree with this change. Basically could read ... as rebranded Something - open Foreman ticket. You'd want to have url/link point as appropriate for the Something.  Needs work to make more configurable the strings, or don't touch.

**Comment by dmitri-d** [comment]
@cliffy94: cheers for the review, I don't think this is going to be accepted though. I'll leave it here for another week, just in case, and will close after that.

**Comment by mccun934** [comment]
@domcleal Why would this be considered a fork?  Did you more-or-less mean you don't want to support maintaining a requirement that all strings showing in the UI stating "Foreman" be extracted to being a configuration variable?

To me this seems no more difficult than localization which is way more invasive

**Comment by dLobatog** [comment]
@witlessbird Ping?

**Comment by dLobatog** [comment]
Closing as it appears that is not going to be merged any soon and seems abandoned. Sorry @witlessbird ! Feel free to reopen and update it if you want to restart the discussion.

**Pull request closed without merging**